nekoyume/Assets/_Scripts/UI/Rank.cs

```diff
@@ -265,12 +266,124 @@ private async void UpdateCategoryAsync(RankCategory category, bool toggleOn)
 
                     rankScroll.Show(mimisbrunnrRankingInfos, true);
                     break;
-                case RankCategory.Weapon:
-                    var weaponRankingInfos = SharedModel.WeaponRankingModel;
-                    if (weaponRankingInfos is null)
+                case RankCategory.Crafting:
+                    if (!isApiLoaded)
+                    {
+                        break;
+                    }
+
+                    var craftRankingInfos = SharedModel.CraftRankingInfos;
+                    if (SharedModel.AgentCraftRankingInfos
+                        .TryGetValue(states.CurrentAvatarKey, out var craftingInfo))
+                    {
+                        myInfoCell.SetDataAsCrafting(craftingInfo);
+                    }
+                    else
+                    {
+                        myInfoCell.SetEmpty(states.CurrentAvatarState);
+                    }
+
+                    rankScroll.Show(craftRankingInfos, true);
+                    break;
```

Comment:
The `craftRankingInfos` variable is necessary?

```suggestion
                case RankCategory.Crafting:
                    if (!isApiLoaded)
                    {
                        break;
                    }

                    if (SharedModel.AgentCraftRankingInfos
                        .TryGetValue(states.CurrentAvatarKey, out var craftingInfo))
                    {
                        myInfoCell.SetDataAsCrafting(craftingInfo);
                    }
                    else
                    {
                        myInfoCell.SetEmpty(states.CurrentAvatarState);
                    }

                    rankScroll.Show(SharedModel.CraftRankingInfos, true);
                    break;
```

Reply:
No, it's not necessary. Fixed it with similar codes!